common/src/ux/codec.rs

```diff
@@ -1,10 +1,24 @@
 use alloc::{string::String, vec::Vec};
 use core::convert::TryInto;
 
-pub trait Serializable: Sized {
+// a reduced-functionality version of Serializable, only used for &str
+// and its composite types, and slices of Serializable types.
+pub trait MiniSerializable {
```

Comment:
I agree that this name is bad.
Maybe since serializing and deserializing are 'peer' behaviours, it should just be
- `Serializable`: `get_serialized_length` and `serialize`
- `Deserializable`: `deserialize`

And anything that currently implements `Serializable` should instead implement both.

Feel free to ignore, I can always do it in a follow up PR.